File: 2º/PDOO/ProyectoJAVA/src/civitas/S_salirCarcel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author nikitastetskiy
 */
public class S_salirCarcel extends Sorpresa{
 
    private MazoSorpresas mazo;
    
    S_salirCarcel(MazoSorpresas mazo, String texto){
        super(texto);
        this.mazo = mazo;
    }
    
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if(jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            boolean nlt = true;
            for(int i=0; i<todos.size(); i++){
                if(todos.get(i).tieneSalvoconducto())
                    nlt = false;
            }
            if(nlt)
                todos.get(actual).obtenerSalvoconducto(this);
            salirDelMazo();
        }
    }
    
    void salirDelMazo() {
        mazo.inhabilitarCartaEspecial(this);
    }
    
    void usada() {
        mazo.habilitarCartaEspecial(this);
    }
    
    @Override
    public String toString() {
       return super.toString();
    }
}

File: 2º/PDOO/ProyectoRUBY/lib/s_salir_carcel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative"sorpresa.rb"
require_relative"mazo_sorpresas.rb"
require_relative"jugador.rb"

module Civitas
  class S_salirCarcel<Sorpresa
    def initialize(mazo)
      super("Sales de la carcel")
      @mazo=mazo
    end
    def aplicarAJugador(actual, todos)
      super.informe(actual, todos)
      boolean salvado=false
      for i in todos.size()
        salvado=todos[i].tieneSalvoconducto()
      end
      if(salvado)then
        todos[actual].obtenerSalvoconducto(this)
        this.salirDelMazo()
      end
    end
    def usada
      @mazito.habilitarCartaEspecial(this)
    end  
    def to_s
      aux=super
      return aux
    end  
  end
end